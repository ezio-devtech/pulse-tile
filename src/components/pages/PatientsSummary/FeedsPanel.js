import React from 'react';

const FeedsPanel = props => {
    return null;
};

export default FeedsPanel;
