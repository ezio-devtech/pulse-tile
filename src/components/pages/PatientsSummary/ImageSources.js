import React from 'react';

export const allergiesPrevImage = null;
export const problemsPrevImage = null;
export const contactsPrevImage = null;
export const medicationsPrevImage = null;
