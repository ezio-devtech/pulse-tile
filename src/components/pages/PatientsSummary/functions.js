import { get } from 'lodash';
import { themeConfigs } from '../../../themes.config';

/**
 * This function returns initial panels number
 * Default number is 4: Allergies, Medications, Contacts, Problems
 *
 * @return {number}
 */
function getInitialPanelsNumber() {
  const defaultPanelsNumber = 4;
  const hiddenCorePlugins = get(themeConfigs, 'corePluginsToHide', []);
  return defaultPanelsNumber - hiddenCorePlugins.length;
}


/**
 * This function return the number of synopsis panels for none-core plugins
 *
 * @return {number}
 */
function getNonCorePanelsNumber(testStoreContent) {
  let result = 0;
  const plugins = Object.keys(testStoreContent);
  plugins.forEach(item => {
    if (item !== 'fetchFeedsRequest') {
      result++;
    }
  });
  return result;
}

/**
 * This function return the total number of synopsis panels
 *
 * @param testStoreContent
 * @return {number}
 */
export function getPanelsNumber(testStoreContent) {
  const initialNumber = getInitialPanelsNumber();
  const pluginsNumber = getNonCorePanelsNumber(testStoreContent);
  return (pluginsNumber > 0) ? (initialNumber + pluginsNumber) : initialNumber;
}

/**
 * This function counts headings blocks number
 * - it equals 1 by default (SHOW)
 * - if theme setting patientsSummaryHasPreviewSettings is TRUE we have additional block (VIEW OF BOARDS)
 *
 * @return {number}
 */
export function getHeadingsNumber() {
    return themeConfigs.patientsSummaryHasPreviewSettings ? 2 : 1;
}