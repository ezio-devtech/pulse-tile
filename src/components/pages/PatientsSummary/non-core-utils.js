import React from 'react';

export const pluginFeedsOnMount = {};
