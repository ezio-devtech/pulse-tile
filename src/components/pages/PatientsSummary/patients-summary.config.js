import React from 'react';
import { get } from 'lodash';
import { themePatientSummaryConfig } from '../../theme/config/plugins';
import {
  allergiesPrevImage,
  problemsPrevImage,
  contactsPrevImage,
  medicationsPrevImage,
} from './ImageSources';
import { themeConfigs } from '../../../themes.config';
import { isPluginVisible, rangePlugins } from '../../../utils/themeSettings-helper';

const problemsTitle = get(themeConfigs.patientsSummaryTitles, 'diagnoses', 'Problems / Diagnosis');
const contactsTitle = get(themeConfigs.patientsSummaryTitles, 'contacts', 'Contacts');
const allergiesTitle = get(themeConfigs.patientsSummaryTitles, 'allergies', 'Allergies');
const medicationsTitle = get(themeConfigs.patientsSummaryTitles, 'medications', 'Medications');

const corePatientsSummaryConfig = [
  {
    key: 'problems',
    panelId: 'summary-panel-problems',
    title: problemsTitle,
    emptyMessage: 'No information available',
    state: 'diagnoses',
    titleCheckboxes: problemsTitle,
    nameCheckboxes: 'problems',
    imgPreview: problemsPrevImage,
    isDefaultSelected: true,
  }, {
    key: 'contacts',
    panelId: 'summary-panel-contacts',
    title: contactsTitle,
    emptyMessage: 'No information available',
    titleCheckboxes: contactsTitle,
    state: 'contacts',
    nameCheckboxes: 'contacts',
    imgPreview: contactsPrevImage,
    isDefaultSelected: true,
  }, {
    key: 'allergies',
    panelId: 'summary-panel-allergies',
    title: allergiesTitle,
    emptyMessage: 'No information available',
    titleCheckboxes: allergiesTitle,
    state: 'allergies',
    nameCheckboxes: 'allergies',
    imgPreview: allergiesPrevImage,
    isDefaultSelected: true,
  }, {
    key: 'medications',
    panelId: 'summary-panel-medications',
    title: medicationsTitle,
    emptyMessage: 'No information available',
    titleCheckboxes: medicationsTitle,
    state: 'medications',
    nameCheckboxes: 'medications',
    imgPreview: medicationsPrevImage,
    isDefaultSelected: true,
  },
];

/**
 * This constant returns list of pattient summary plugins, excluded corePluginsToHide (themes settings)
 *
 * @return {array}
 */
const filterPatientsSummaryConfig = corePatientsSummaryConfig.filter(item => {
  const hiddenCorePlugins = get(themeConfigs, 'corePluginsToHide', []);
  return isPluginVisible(hiddenCorePlugins, item.state);
});
const totalSummaryConfig = filterPatientsSummaryConfig.concat(themePatientSummaryConfig);

export const patientsSummaryConfig = rangePlugins(totalSummaryConfig);

export const defaultViewOfBoardsSelected = {
  full: true,
  preview: false,
  list: false,
};
