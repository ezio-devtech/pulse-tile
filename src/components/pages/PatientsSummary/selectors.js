import _ from 'lodash/fp';
import { createSelector } from 'reselect';
import { themeSynopsisSelector } from '../../theme/config/themeSelectors';

function getCommonSelector(patientProblemsSelector, patientContactsSelector, patientAllergiesSelector, patientMedicationsSelector, themeSynopsisSelector) {
  let result = createSelector(
      patientProblemsSelector,
      patientContactsSelector,
      patientAllergiesSelector,
      patientMedicationsSelector,
      (
        problems,
        contacts,
        allergies,
        medications,
      ) => {
        return {
          boards: {
            problems: problems,
            contacts: contacts,
            allergies: allergies,
            medications: medications,
          },
        };
      }
    );
    if (themeSynopsisSelector) {
      result = createSelector(
        patientProblemsSelector,
        patientContactsSelector,
        patientAllergiesSelector,
        patientMedicationsSelector,
        themeSynopsisSelector,
        (
          problems,
          contacts,
          allergies,
          medications,
          theme,
        ) => {
          const coreResult = {
            problems: problems,
            contacts: contacts,
            allergies: allergies,
            medications: medications,
          };
          const themeResult = theme;
          const totalResult = Object.assign(coreResult, themeResult);
          return {
            boards: totalResult,
            };
          }
        );
    }
    return result;
}

const patientProblemsSelector = createSelector(
  ({ patientsDiagnoses }) => patientsDiagnoses,
  (state, props) => _.getOr(null, 'match.params.userId', props),
  (patientsDiagnoses, userId) => {
    let problems = {};
    if (patientsDiagnoses[userId]) {
      problems = patientsDiagnoses[userId];
      } else {
        problems = [{text: 'Loading ...'}, '', '', ''];
      }
      return problems;
  }
);

const patientContactsSelector = createSelector(
  ({ patientsContacts }) => patientsContacts,
  (state, props) => _.getOr(null, 'match.params.userId', props),
  (patientsContacts, userId) => {
    let contacts = {};
    if (patientsContacts[userId]) {
      contacts = patientsContacts[userId];
      } else {
        contacts = [{text: 'Loading ...'}, '', '', ''];
      }
      return contacts;
  }
);

const patientAllergiesSelector = createSelector(
  ({ patientsAllergies }) => patientsAllergies,
  (state, props) => _.getOr(null, 'match.params.userId', props),
  (patientsAllergies, userId) => {
    let allergies = {};
    if (patientsAllergies[userId]) {
      allergies = patientsAllergies[userId];
    } else {
      allergies = [{text: 'Loading ...'}, '', '', ''];
    }
    return allergies;
  }
);

const patientMedicationsSelector = createSelector(
  ({ patientsMedications }) => patientsMedications,
  (state, props) => _.getOr(null, 'match.params.userId', props),
  (patientsMedications, userId) => {
    let medications = {};
    if (patientsMedications[userId]) {
      medications = patientsMedications[userId];
    } else {
      medications = [{text: 'Loading ...'}, '', '', ''];
    }
    return medications;
  }
);

export const summarySynopsisSelector = getCommonSelector(
  patientProblemsSelector,
  patientContactsSelector,
  patientAllergiesSelector,
  patientMedicationsSelector,
  themeSynopsisSelector
);
