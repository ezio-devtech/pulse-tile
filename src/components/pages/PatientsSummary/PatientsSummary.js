import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'react-bootstrap';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { compose, lifecycle } from 'recompose';
import { get } from 'lodash';
import { themeConfigs } from '../../../themes.config';
import SimpleDashboardPanel from './SimpleDashboardPanel';
import ConfirmationModal from '../../ui-elements/ConfirmationModal/ConfirmationModal';
import PatientsSummaryListHeader from './header/PatientsSummaryListHeader';
import { summarySynopsisSelector } from './selectors';
import { patientsSummaryConfig, defaultViewOfBoardsSelected } from './patients-summary.config';
import { fetchPatientDiagnosesSynopsisRequest } from '../Diagnosis/ducks/fetch-patient-diagnoses.duck';
import { fetchPatientContactsSynopsisRequest } from '../Contacts/ducks/fetch-patient-contacts.duck';
import { fetchPatientAllergiesSynopsisRequest } from '../Allergies/ducks/fetch-patient-allergies.duck';
import { fetchPatientMedicationsSynopsisRequest } from '../Medications/ducks/fetch-patient-medications.duck';

import {
  fetchPatientProblemsSynopsisOnMount,
  fetchPatientContactsSynopsisOnMount,
  fetchPatientAllergiesSynopsisOnMount,
  fetchPatientMedicationsSynopsisOnMount,
} from '../../../utils/HOCs/fetch-patients.utils';

import ExtraPatientsSummaryPanels from '../../theme/components/ExtraPatientsSummaryPanels';

import { themeSynopsisOnMount, themeSynopsisRequests } from '../../theme/config/synopsisRequests';
import { dashboardVisible, dashboardBeing } from '../../../plugins.config';
import { testConstants, isDevMode } from '../../../config/for-test.constants';

const coreActionsArray = {
  fetchPatientDiagnosesSynopsisRequest,
  fetchPatientContactsSynopsisRequest,
  fetchPatientAllergiesSynopsisRequest,
  fetchPatientMedicationsSynopsisRequest,
};
const actionsArray = Object.assign(coreActionsArray, themeSynopsisRequests);
const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(actionsArray, dispatch)
});

@connect(summarySynopsisSelector, mapDispatchToProps)

@compose(
  lifecycle(fetchPatientProblemsSynopsisOnMount),
  lifecycle(fetchPatientContactsSynopsisOnMount),
  lifecycle(fetchPatientAllergiesSynopsisOnMount),
  lifecycle(fetchPatientMedicationsSynopsisOnMount),
  lifecycle(themeSynopsisOnMount),
)

export default class PatientsSummary extends PureComponent {
  static propTypes = {
    boards: PropTypes.shape({}).isRequired,
  };

  static contextTypes = {
    router: PropTypes.shape({
      history: PropTypes.object,
    }),
  };

  state = {
    selectedCategory: [],
    selectedViewOfBoards: defaultViewOfBoardsSelected,
    isDisclaimerModalVisible: false,
    isCategory: {},
  };

  componentWillMount() {
    const isShowDisclaimerOfRedirect = localStorage.getItem('isShowDisclaimerOfRedirect');
    localStorage.removeItem('isShowDisclaimerOfRedirect');

    if (isShowDisclaimerOfRedirect) {
      this.setState({ isDisclaimerModalVisible: true });
    }

    this.setState({ selectedCategory: this.getDefaultCategorySelected() });
  }

  getDefaultCategorySelected = () => {
    const defaultCategorySelected = {};

    patientsSummaryConfig.forEach((item) => {
      if (dashboardVisible[item.key] !== undefined) {
        defaultCategorySelected[item.key] = dashboardVisible[item.key];
      } else {
        defaultCategorySelected[item.key] = item.isDefaultSelected;
      }
    });

    return defaultCategorySelected;
  };

  closeDisclaimer = () => this.setState({ isDisclaimerModalVisible: false });

  handleCategorySelected = selectedCategory => this.setState({ selectedCategory });

  handleViewOfBoardsSelected = selectedViewOfBoards => this.setState({ selectedViewOfBoards });

  handleGoToState = (state, externalTransitionUrl) => {
    if (state.indexOf('http://') !== -1 ||
      state.indexOf('https://') !== -1 ||
      state.indexOf('www.') !== -1) {
      if (externalTransitionUrl) {
        window.open(externalTransitionUrl)
      } else {
        window.open(state)
      }
    } else {
      this.context.router.history.push(state)
    }
  };

  render() {

    const { boards } = this.props;

    const { selectedCategory, selectedViewOfBoards, isDisclaimerModalVisible, isCategory } = this.state;
    let isHasPreview = selectedViewOfBoards.full || selectedViewOfBoards.preview;
    const isHasList = selectedViewOfBoards.full || selectedViewOfBoards.list;

    if (!themeConfigs.patientsSummaryHasPreviewSettings) { isHasPreview = false; }

    return (<section className="page-wrapper">
      <Row>
        <Col xs={12}>
          <div className="panel panel-primary panel-dashboard">
            <PatientsSummaryListHeader
              onCategorySelected={this.handleCategorySelected}
              onViewOfBoardsSelected={this.handleViewOfBoardsSelected}
              selectedCategory={selectedCategory}
              selectedViewOfBoards={selectedViewOfBoards}
              title={themeConfigs.patientsSummaryPageName}
              boards={boards}
            />
            <div className="panel-body">
              <div className="dashboard">

                {patientsSummaryConfig.map((item, index) => {
                  const imageSource = isDevMode ? (testConstants.hostName + item.imgPreview) : item.imgPreview;
                  const items = boards[item.key];
                  const loadingItems = (items.length > 0) ? items : [{text: item.emptyMessage}];
                  return (selectedCategory[item.key] && dashboardBeing[item.key] !== false ?
                    <SimpleDashboardPanel
                      id={item.panelId}
                      key={index}
                      title={item.title}
                      items={loadingItems}
                      state={item.state}
                      goToState={this.handleGoToState}
                      srcPrevirew={imageSource}
                      isHasPreview={isHasPreview}
                      isHasList={isHasList}
                    />
                    : null)
                })}

                <ExtraPatientsSummaryPanels
                  boards={boards}
                  handleGoToState={this.handleGoToState}
                  isHasPreview={isHasPreview}
                  isHasList={isHasList}
                />

              </div>
            </div>
          </div>
        </Col>
      </Row>
      {isDisclaimerModalVisible && <ConfirmationModal
        title={'Notification'}
        isShow
        onOk={this.closeDisclaimer}
        onHide={this.closeDisclaimer}
        isShowOkButton
      >
        <span>You was redirected to your home page because you are logged in as a PHR user.</span>
      </ConfirmationModal>}
    </section>)
  }
}
