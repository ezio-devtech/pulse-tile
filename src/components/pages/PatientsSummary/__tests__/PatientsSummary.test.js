import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';
import { get } from 'lodash';

import { testStoreContent } from '../../../theme/config/plugins';
import PatientsSummary from '../PatientsSummary';
import { themeConfigs } from '../../../../themes.config';
import { getPanelsNumber } from '../functions';

Enzyme.configure({ adapter: new Adapter() });

class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}
global.localStorage = new LocalStorageMock();

const mockStore = configureStore();

const coreStoreContent = {
    userId: '9999999000',
    patientsDiagnoses: {},
    patientsContacts: {},
    patientsAllergies: {},
    patientsMedications: {},
    feeds: [{
      name: 'Leeds Live - Whats on',
      landingPageUrl: 'https://www.leeds-live.co.uk/best-in-leeds/whats-on-news/',
      rssFeedUrl: 'https://www.leeds-live.co.uk/best-in-leeds/whats-on-news/?service=rss',
      sourceId: 'testSourceID4',
    }, {
      name: 'Leeds CC Local News',
      landingPageUrl: 'https://news.leeds.gov.uk',
      rssFeedUrl: 'https://news.leeds.gov.uk/tagfeed/en/tags/Leeds-news',
      sourceId: 'testSourceID5',
    }],
};
const storeContent = Object.assign(coreStoreContent, testStoreContent);

const store = mockStore(storeContent);
const match = {
  params: {},
};
const location = {
  pathname: '/patients',
  search: '?ageRange=61-80',
};
const context = {
  router: {
    history: {
      push: () => {},
      replace: () => {},
      location: {
        pathname: '/patients',
        search: '?ageRange=61-80',
      },
    },
    route: {
      match: {
        params: {},
      },
    },
  },
};

const testProps = {
  onCategorySelected: () => {},
  selectedCategory: {
    problems: true,
    contacts: true,
    allergies: true,
    medications: true,
  },
};

describe('Component <PatientsSummary />', () => {
  it('should renders with all props correctly', () => {
    let component = shallow(
      <PatientsSummary
        store={store}
        match={match}
        location={location}
        onCategorySelected={testProps.onCategorySelected}
        selectedCategory={testProps.selectedCategory}
      />, { context })
        .dive()
        .dive()
        .dive()
        .dive()
        .dive()
        .dive();

    expect(component).toMatchSnapshot();

    expect(component.instance().props.onCategorySelected).toEqual(testProps.onCategorySelected);
    expect(component.instance().props.selectedCategory).toEqual(testProps.selectedCategory);

    expect(component.find('.page-wrapper')).toHaveLength(1);
    expect(component.find('PatientsSummaryListHeader')).toHaveLength(1);
    expect(component.find('.dashboard')).toHaveLength(1);

    const panelsNumber = getPanelsNumber(testStoreContent);

    expect(component.find('SimpleDashboardPanel')).toHaveLength(panelsNumber);
    expect(component.find('ConfirmationModal')).toHaveLength(0);

    component.instance().handleGoToState('contacts');

    component.instance().handleCategorySelected({
      problems: false,
      contacts: false,
      allergies: false,
      medications: false,
    });
    component.setState({ selectedCategory: {
      problems: false,
      contacts: false,
      allergies: false,
      medications: false,
    } });

    component.instance().handleViewOfBoardsSelected({ full: true });
    component.instance().handleViewOfBoardsSelected({ preview: true });

    expect(component.find('SimpleDashboardPanel')).toHaveLength(0);

    themeConfigs.isLeedsPHRTheme = true;
    themeConfigs.patientsSummaryHasPreviewSettings = true;
    component.setState({ selectedCategory: { 'nytimes_com': true } });
    expect(component.find('SimpleDashboardPanel')).toHaveLength(0);

    component.instance().handleGoToState('http://www.bbc.co.uk/news/health');
    component.instance().handleGoToState('http://www.bbc.co.uk/news/health', 'http://testUrl1');
  });

  it('should renders with Disclaimer Modal correctly', () => {
    localStorage.setItem('isShowDisclaimerOfRedirect', true);
    let component = shallow(
      <PatientsSummary
        store={store}
        match={match}
        location={location}
        onCategorySelected={testProps.onCategorySelected}
        selectedCategory={testProps.selectedCategory}
      />, { context })
        .dive()
        .dive()
        .dive()
        .dive()
        .dive()
        .dive();

    expect(component).toMatchSnapshot();

    expect(component.find('ConfirmationModal')).toHaveLength(1);
    component.instance().closeDisclaimer();
    expect(component.find('ConfirmationModal')).toHaveLength(1);
  });

  it('should renders Feeds correctly', () => {
    themeConfigs.isLeedsPHRTheme = true;
    const component = shallow(
      <PatientsSummary
        store={store}
        match={match}
        location={location}
        onCategorySelected={testProps.onCategorySelected}
        selectedCategory={testProps.selectedCategory}
      />, { context })
        .dive()
        .dive()
        .dive()
        .dive()
        .dive()
        .dive();

    expect(component).toMatchSnapshot();
  });

});

