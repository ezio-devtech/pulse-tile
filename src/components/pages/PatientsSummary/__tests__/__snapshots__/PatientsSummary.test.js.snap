// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <PatientsSummary /> should renders Feeds correctly 1`] = `
<section
  className="page-wrapper"
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}
    >
      <div
        className="panel panel-primary panel-dashboard"
      >
        <PatientsSummaryListHeader
          boards={
            Object {
              "allergies": Array [
                Object {
                  "text": "Loading ...",
                },
                "",
                "",
                "",
              ],
              "contacts": Array [
                Object {
                  "text": "Loading ...",
                },
                "",
                "",
                "",
              ],
              "medications": Array [
                Object {
                  "text": "Loading ...",
                },
                "",
                "",
                "",
              ],
              "problems": Array [
                Object {
                  "text": "Loading ...",
                },
                "",
                "",
                "",
              ],
            }
          }
          onCategorySelected={[Function]}
          onViewOfBoardsSelected={[Function]}
          selectedCategory={
            Object {
              "allergies": true,
              "contacts": true,
              "medications": true,
              "problems": true,
            }
          }
          selectedViewOfBoards={
            Object {
              "full": true,
              "list": false,
              "preview": false,
            }
          }
          title="Patient Summary"
        />
        <div
          className="panel-body"
        >
          <div
            className="dashboard"
          >
            <SimpleDashboardPanel
              goToState={[Function]}
              id="summary-panel-problems"
              isFeeds={false}
              isHasList={true}
              isHasPreview={true}
              items={
                Array [
                  Object {
                    "text": "Loading ...",
                  },
                  "",
                  "",
                  "",
                ]
              }
              key="0"
              srcPrevirew={null}
              state="diagnoses"
              title="Problems / Diagnosis"
            />
            <SimpleDashboardPanel
              goToState={[Function]}
              id="summary-panel-medications"
              isFeeds={false}
              isHasList={true}
              isHasPreview={true}
              items={
                Array [
                  Object {
                    "text": "Loading ...",
                  },
                  "",
                  "",
                  "",
                ]
              }
              key="1"
              srcPrevirew={null}
              state="medications"
              title="Medications"
            />
            <SimpleDashboardPanel
              goToState={[Function]}
              id="summary-panel-allergies"
              isFeeds={false}
              isHasList={true}
              isHasPreview={true}
              items={
                Array [
                  Object {
                    "text": "Loading ...",
                  },
                  "",
                  "",
                  "",
                ]
              }
              key="2"
              srcPrevirew={null}
              state="allergies"
              title="Allergies"
            />
            <SimpleDashboardPanel
              goToState={[Function]}
              id="summary-panel-contacts"
              isFeeds={false}
              isHasList={true}
              isHasPreview={true}
              items={
                Array [
                  Object {
                    "text": "Loading ...",
                  },
                  "",
                  "",
                  "",
                ]
              }
              key="3"
              srcPrevirew={null}
              state="contacts"
              title="Contacts"
            />
            <ExtraPatientsSummarySelectors
              boards={
                Object {
                  "allergies": Array [
                    Object {
                      "text": "Loading ...",
                    },
                    "",
                    "",
                    "",
                  ],
                  "contacts": Array [
                    Object {
                      "text": "Loading ...",
                    },
                    "",
                    "",
                    "",
                  ],
                  "medications": Array [
                    Object {
                      "text": "Loading ...",
                    },
                    "",
                    "",
                    "",
                  ],
                  "problems": Array [
                    Object {
                      "text": "Loading ...",
                    },
                    "",
                    "",
                    "",
                  ],
                }
              }
              handleGoToState={[Function]}
              isHasList={true}
              isHasPreview={true}
            />
          </div>
        </div>
      </div>
    </Col>
  </Row>
</section>
`;

exports[`Component <PatientsSummary /> should renders with Disclaimer Modal correctly 1`] = `
<section
  className="page-wrapper"
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}
    >
      <div
        className="panel panel-primary panel-dashboard"
      >
        <PatientsSummaryListHeader
          boards={
            Object {
              "allergies": Array [
                Object {
                  "text": "Loading ...",
                },
                "",
                "",
                "",
              ],
              "contacts": Array [
                Object {
                  "text": "Loading ...",
                },
                "",
                "",
                "",
              ],
              "medications": Array [
                Object {
                  "text": "Loading ...",
                },
                "",
                "",
                "",
              ],
              "problems": Array [
                Object {
                  "text": "Loading ...",
                },
                "",
                "",
                "",
              ],
            }
          }
          onCategorySelected={[Function]}
          onViewOfBoardsSelected={[Function]}
          selectedCategory={
            Object {
              "allergies": true,
              "contacts": true,
              "medications": true,
              "problems": true,
            }
          }
          selectedViewOfBoards={
            Object {
              "full": true,
              "list": false,
              "preview": false,
            }
          }
          title="Patient Summary"
        />
        <div
          className="panel-body"
        >
          <div
            className="dashboard"
          >
            <SimpleDashboardPanel
              goToState={[Function]}
              id="summary-panel-problems"
              isFeeds={false}
              isHasList={true}
              isHasPreview={true}
              items={
                Array [
                  Object {
                    "text": "Loading ...",
                  },
                  "",
                  "",
                  "",
                ]
              }
              key="0"
              srcPrevirew={null}
              state="diagnoses"
              title="Problems / Diagnosis"
            />
            <SimpleDashboardPanel
              goToState={[Function]}
              id="summary-panel-medications"
              isFeeds={false}
              isHasList={true}
              isHasPreview={true}
              items={
                Array [
                  Object {
                    "text": "Loading ...",
                  },
                  "",
                  "",
                  "",
                ]
              }
              key="1"
              srcPrevirew={null}
              state="medications"
              title="Medications"
            />
            <SimpleDashboardPanel
              goToState={[Function]}
              id="summary-panel-allergies"
              isFeeds={false}
              isHasList={true}
              isHasPreview={true}
              items={
                Array [
                  Object {
                    "text": "Loading ...",
                  },
                  "",
                  "",
                  "",
                ]
              }
              key="2"
              srcPrevirew={null}
              state="allergies"
              title="Allergies"
            />
            <SimpleDashboardPanel
              goToState={[Function]}
              id="summary-panel-contacts"
              isFeeds={false}
              isHasList={true}
              isHasPreview={true}
              items={
                Array [
                  Object {
                    "text": "Loading ...",
                  },
                  "",
                  "",
                  "",
                ]
              }
              key="3"
              srcPrevirew={null}
              state="contacts"
              title="Contacts"
            />
            <ExtraPatientsSummarySelectors
              boards={
                Object {
                  "allergies": Array [
                    Object {
                      "text": "Loading ...",
                    },
                    "",
                    "",
                    "",
                  ],
                  "contacts": Array [
                    Object {
                      "text": "Loading ...",
                    },
                    "",
                    "",
                    "",
                  ],
                  "medications": Array [
                    Object {
                      "text": "Loading ...",
                    },
                    "",
                    "",
                    "",
                  ],
                  "problems": Array [
                    Object {
                      "text": "Loading ...",
                    },
                    "",
                    "",
                    "",
                  ],
                }
              }
              handleGoToState={[Function]}
              isHasList={true}
              isHasPreview={true}
            />
          </div>
        </div>
      </div>
    </Col>
  </Row>
  <ConfirmationModal
    isShow={true}
    isShowCancelButton={false}
    isShowOkButton={true}
    onHide={[Function]}
    onOk={[Function]}
    textCancelButton="Cancel"
    textOkButton="Ok"
    title="Notification"
  >
    <span>
      You was redirected to your home page because you are logged in as a PHR user.
    </span>
  </ConfirmationModal>
</section>
`;

exports[`Component <PatientsSummary /> should renders with all props correctly 1`] = `
<section
  className="page-wrapper"
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}
    >
      <div
        className="panel panel-primary panel-dashboard"
      >
        <PatientsSummaryListHeader
          boards={
            Object {
              "allergies": Array [
                Object {
                  "text": "Loading ...",
                },
                "",
                "",
                "",
              ],
              "contacts": Array [
                Object {
                  "text": "Loading ...",
                },
                "",
                "",
                "",
              ],
              "medications": Array [
                Object {
                  "text": "Loading ...",
                },
                "",
                "",
                "",
              ],
              "problems": Array [
                Object {
                  "text": "Loading ...",
                },
                "",
                "",
                "",
              ],
            }
          }
          onCategorySelected={[Function]}
          onViewOfBoardsSelected={[Function]}
          selectedCategory={
            Object {
              "allergies": true,
              "contacts": true,
              "medications": true,
              "problems": true,
            }
          }
          selectedViewOfBoards={
            Object {
              "full": true,
              "list": false,
              "preview": false,
            }
          }
          title="Patient Summary"
        />
        <div
          className="panel-body"
        >
          <div
            className="dashboard"
          >
            <SimpleDashboardPanel
              goToState={[Function]}
              id="summary-panel-problems"
              isFeeds={false}
              isHasList={true}
              isHasPreview={false}
              items={
                Array [
                  Object {
                    "text": "Loading ...",
                  },
                  "",
                  "",
                  "",
                ]
              }
              key="0"
              srcPrevirew={null}
              state="diagnoses"
              title="Problems / Diagnosis"
            />
            <SimpleDashboardPanel
              goToState={[Function]}
              id="summary-panel-medications"
              isFeeds={false}
              isHasList={true}
              isHasPreview={false}
              items={
                Array [
                  Object {
                    "text": "Loading ...",
                  },
                  "",
                  "",
                  "",
                ]
              }
              key="1"
              srcPrevirew={null}
              state="medications"
              title="Medications"
            />
            <SimpleDashboardPanel
              goToState={[Function]}
              id="summary-panel-allergies"
              isFeeds={false}
              isHasList={true}
              isHasPreview={false}
              items={
                Array [
                  Object {
                    "text": "Loading ...",
                  },
                  "",
                  "",
                  "",
                ]
              }
              key="2"
              srcPrevirew={null}
              state="allergies"
              title="Allergies"
            />
            <SimpleDashboardPanel
              goToState={[Function]}
              id="summary-panel-contacts"
              isFeeds={false}
              isHasList={true}
              isHasPreview={false}
              items={
                Array [
                  Object {
                    "text": "Loading ...",
                  },
                  "",
                  "",
                  "",
                ]
              }
              key="3"
              srcPrevirew={null}
              state="contacts"
              title="Contacts"
            />
            <ExtraPatientsSummarySelectors
              boards={
                Object {
                  "allergies": Array [
                    Object {
                      "text": "Loading ...",
                    },
                    "",
                    "",
                    "",
                  ],
                  "contacts": Array [
                    Object {
                      "text": "Loading ...",
                    },
                    "",
                    "",
                    "",
                  ],
                  "medications": Array [
                    Object {
                      "text": "Loading ...",
                    },
                    "",
                    "",
                    "",
                  ],
                  "problems": Array [
                    Object {
                      "text": "Loading ...",
                    },
                    "",
                    "",
                    "",
                  ],
                }
              }
              handleGoToState={[Function]}
              isHasList={true}
              isHasPreview={false}
            />
          </div>
        </div>
      </div>
    </Col>
  </Row>
</section>
`;
